### PR TITLE
expression/aggregation: migrate test-infra to testify for base_func_test.go

### LINKED ISSUE #27907: migrate test-infra to testify for `expression/aggregation/base_func_test.go`
This is a subtask of #27904

### PR DESCRIPTION
**Pull request by penthaapatel**
Signed-off-by: Penthaa Patel <penthaapatel@gmail.com>
### What problem does this PR solve?

Issue Number: close #27907 

### What is changed and how it works?
What's Changed:

- migrate test-infra to testify for `expression/aggregation/base_func_test.go`

### Release note 
```release-note
None
```

### CODE REVIEW

**Review by xiongjiwei — APPROVED** [review]

File: expression/aggregation/base_func_test.go [hunk 1]
```diff
@@ -1,41 +1,44 @@
...
+type mockBaseFuncSuite struct {
 	ctx sessionctx.Context
 }
 
-func (s *testBaseFuncSuite) SetUpSuite(c *check.C) {
+func createBaseFuncSuite(t *testing.T) (s *mockBaseFuncSuite) {
+	s = new(mockBaseFuncSuite)
 	s.ctx = mock.NewContext()
+	return
 }
```
Comment (tisonkun) [inline expression/aggregation/base_func_test.go:23]: This suite is nit, you can just inline the logic.

Comment (penthaapatel) [reply expression/aggregation/base_func_test.go]: Fixed it - ed1ada7

Comment (penthaapatel) [reply expression/aggregation/base_func_test.go]: Fixed it - 2aea1c8

**Review by tisonkun — APPROVED** [review]
Thanks for your updates! LGTM.

**Comment by tisonkun** [comment]
@xiongjiwei could you please help on merging this PR?

**Pull request merged**